_plugins/generators/onsite_groups_generator.rb

```diff
@@ -0,0 +1,36 @@
+module Jekyll
+  class OnsiteGroupsGenerator < Generator
```

Comment:
This is interesting. One thing I love about Jekyll is that you can solve the same problem in so many different ways.  I was thinking about this entirely different, but this gets the job done.

Thinking ahead, you'll also need another reference to these locations when building out the cards at the bottom of the OSG landing page. Are you planning to amend this generator at that time? Or have you thought about another approach for that one?